src/librustc_lint/late.rs

```diff
@@ -31,6 +30,12 @@ use syntax::walk_list;
 use log::debug;
 use std::slice;
 
+/// Extract the `LintStore` from the query context.
+/// This function exists because we've erased `LintStore` as `dyn Any` in the context.
+crate fn unerased_lint_store<'tcx>(tcx: TyCtxt<'tcx>) -> &'tcx LintStore {
```

Comment:
I guess it probably doesn't matter but it would be nice to return `&'tcx Lrc<LintStore>` here I think, or, if we don't need that then maybe the Lrc can be dropped? (I forget why it exists).

Reply:
(We concluded in private that `&'tcx LintStore` is fine, as it promises less to callers.)